lib/ansible/modules/clustering/kubevirt/kubevirt_service.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2018, KubeVirt Team <@kubevirt>
+# Apache License, Version 2.0
+# (see LICENSE or http://www.apache.org/licenses/LICENSE-2.0)
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: kubevirt_service
+
+short_description: Manage Services for KubeVirt virtual machines
+
+description:
+    - Use Openshift Python SDK to manage Services for KubeVirt virtual machines.
+
+version_added: "2.8"
+
+author: KubeVirt Team (@kubevirt)
+
+options:
+    state:
+        description:
+            - Determines if an object should be created, patched, or deleted. When set to C(present), an object will be created, if it does not already exist. If set to C(absent), an existing object will be deleted. If set to C(present), an existing object will be patched, if its attributes differ from those specified.
+        default: present
+        choices:
+            - present
+            - absent
+        type: str
+    name:
+        description:
+            - Use to specify a Service object name.
+        required: true
+        type: str
+    namespace:
+        description:
+            - Use to specify a Service object namespace.
+        required: true
+        type: str
+    selector:
+        description:
+            - TODO
+        type: dict
+    type:
+        description:
+            - TODO
+        default: ClusterIP
+        choices:
+            - NodePort
+            - ClusterIP
+            - LoadBalancer
+    ports:
+        description:
+            - TODO
+        required: true
+        type: list
+    labels:
```

Comment:
I would go with default here `[]`

Reply:
Of course.

Reply:
I think you thought `{}`. But also we must be careful with this as now if there is a service existing and you run this module and don't define any label, it will remove all labels from that service.

Reply:
Yeah, I saw list above but indeed I meant dictionary.

Reply:
@machacekondra well yes, but what other options do we have? If my definition has for example two labels, then an update adds those labels. If I remove them from my yaml and then update, then all labels get cleared. I don't think we're supposed to support a case where k8s objects are partially maintained by our ansible module and partially by something else. Or do I not know something about how k8s modules are supposed to work?

Reply:
What I just wanted to say is that having `None` as default in this case is more safe IMHO.

Reply:
Dropped `labels:` entirely, so no longer an issue.